src/main/java/io/kestra/plugin/azure/runner/AzureBatchScriptRunner.java

```diff
@@ -0,0 +1,183 @@
+        }
+        boolean hasFilesToDownload = !ListUtils.isEmpty(filesToDownload);
+        if (hasFilesToDownload && !hasBlobStorage) {
+            throw new IllegalArgumentException("You must provide a way to connect to a Blob Storage container to use `outputFiles`");
+        }
+
+        BlobContainerClient blobContainerClient = blobStorage.blobContainerClient(runContext);
+
+        String jobId = IdUtils.create();
+
+        List<ResourceFile> resourceFiles = filesToUpload.stream().map(throwFunction(file -> {
+            // Use path to eventually deduplicate leading '/'
+            String blobName = blobStorageWorkingDirName + Path.of("/" + file);
+            blobContainerClient.getBlobClient(blobName)
+                .uploadFromFile(runContext.resolve(Path.of(file)).toString(), true);
+
+            SharedAccess task = SharedAccess.builder()
+                .id(SharedAccess.class.getSimpleName())
+                .type(io.kestra.plugin.azure.storage.blob.List.class.getName())
+                .endpoint(this.endpoint)
+                .connectionString(blobStorage.getConnectionString())
+                .container(blobStorage.getContainerName())
+                .name(blobName)
+                .expirationDate("{{ now() | dateAdd(1, 'DAYS')  }}")
+                .permissions(Set.of(SharedAccess.Permission.READ))
+                .build();
+
+            SharedAccess.Output sas = task.run(runContext);
+
+            return ResourceFile.builder()
+                .filePath(file.startsWith("/") ? file.substring(1) : file)
+                // Use path to eventually deduplicate leading '/'
+                .httpUrl(sas.getUri().toString())
+                .build();
+        })).toList();
+
+        AbstractLogConsumer logConsumer = commandsWrapper.getLogConsumer();
+        Create createJob = Create.builder()
+            .id("create")
+            .type(Create.class.getName())
+            .account(this.account)
+            .accessKey(this.accessKey)
+            .endpoint(this.endpoint)
+            .poolId(this.poolId)
+            .maxDuration(this.waitUntilCompletion)
+            .job(
+                Job.builder()
+                    .id(jobId)
+                    .build()
+            )
+            .tasks(List.of(
+                Task.builder()
+                    .id("task-" + jobId)
+                    .constraints(
+                        TaskConstraints.builder()
+                            .maxWallClockTime(this.waitUntilCompletion)
+                            .maxTaskRetryCount(0)
+                            .build()
+                    )
+                    .interpreter(command.get(0))
+                    .interpreterArgs(command.size() > 1 ? new String[]{command.get(1)} : new String[0])
+                    .commands(command.size() > 2 ? command.subList(2, command.size()) : Collections.emptyList())
```

Comment:
Everythig should be a command.
I think it's a bug and I forgot to pass the interpreter from the CommandsWrapper ...

Reply:
I'm just dealing with the Create task's mandatory properties... Idk I feel like the Create task is recreating the whole Commands / Scripts system but I have the feeling that reworking it in that PR would be too much